### DISCUSSION TITLE
Tooltips?

### DISCUSSION THREAD

**Discussion by azban** [terrastruct/d2#329] [Feature Requests]
It would be nice to have tooltips for objects, so that by default the diagrams are concise, but more information can be encoded for someone looking to dig into specific elements.

**Comment by azban** [comment]
Just kidding, found this in the UI. :)

**Comment by alixander** [comment]
we still do need to implement that for local SVGs though. made an issue from it, thx @azban !

**Reply by alixander** [reply]
https://github.com/terrastruct/d2/issues/367

**Comment by alixander** [comment]
i've been thinking for a while on how tooltips should behave on exports, when there's no interactivity from a PNG. what if we laid out all the tooltips in like a legend on the side? kind of like this graphic. @nhooyr @gavin-ts @ejulio-ts 

<img width="898" alt="Screen Shot 2022-12-16 at 11 53 07 PM" src="https://user-images.githubusercontent.com/3120367/208231843-2e69cfae-0adb-444d-958f-30b8b666145c.png">

**Reply by ejulio-ts** [reply]
Looks good.
What would be the syntax for it?
```
node: {
  tooltip: |md
# a tooltip
markdown
  |
}
```

**Comment by nhooyr** [comment]
~~I believe that's already the syntax for it. The proposed render seems good to me.~~

nvm: we don't have tooltips what am I saying. The proposed render with a tooltip field like suggested by @ejulio-ts seems good to me.

**Comment by alixander** [comment]
this will be in the next release: https://github.com/terrastruct/d2/pull/545
